[Automation] javac compile fails for org.glassfish.jersey.containers:jersey-container-servlet:4.0.1

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean compileTestJava -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25656145616/job/75305207725
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.glassfish.jersey.containers:jersey-container-servlet:2.34
Raw versions JSON: [
  "4.0.1",
  "4.0.2",
  "5.0.0-M1"
]
Parsed version count: 3
Parsed versions: 4.0.1 4.0.2 5.0.0-M1
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.glassfish.jersey.containers:jersey-container-servlet:2.34:4.0.1
========================================================================================
========================================================================================
 org.glassfish.jersey.containers:jersey-container-servlet:2.34:4.0.1 stage native-image run
========================================================================================
Starting stage 'native-image run' for 4.0.1 at 2026-05-11T07:25:09Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean test -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.glassfish.jersey.containers:jersey-container-servlet:2.34
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                 ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
        Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                              ^
  symbol:   variable LocalizationMessages
  location: class LocalizationMessagesInnerBundleSupplierTest
2 errors

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
          Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                                ^
    symbol:   variable LocalizationMessages
    location: class LocalizationMessagesInnerBundleSupplierTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
  import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                   ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

BUILD FAILED in 4s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.glassfish.jersey.containers:jersey-container-servlet:2.34 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 13s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 4.0.1 at 2026-05-11T07:25:22Z with exit code 1 after 13s
❌ ERROR: 'native-image run' for 4.0.1 failed with exit code 1.
SOFT-FAIL[native-image run][4.0.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean test -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"]
========================================================================================
 org.glassfish.jersey.containers:jersey-container-servlet:2.34:4.0.1 stage javac compile
========================================================================================
Starting stage 'javac compile' for 4.0.1 at 2026-05-11T07:25:22Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean compileTestJava -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 906ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                 ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
        Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                              ^
  symbol:   variable LocalizationMessages
  location: class LocalizationMessagesInnerBundleSupplierTest
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
          Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                                ^
    symbol:   variable LocalizationMessages
    location: class LocalizationMessagesInnerBundleSupplierTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
  import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                   ^
  2 errors

* Try:

> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 4.0.1 at 2026-05-11T07:25:25Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 4.0.1 failed with exit code 1.
FAILED[javac compile][4.0.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean compileTestJava -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean compileTestJava -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25656145616/job/75305207659
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.glassfish.jersey.containers:jersey-container-servlet:2.34
Raw versions JSON: [
  "4.0.1",
  "4.0.2",
  "5.0.0-M1"
]
Parsed version count: 3
Parsed versions: 4.0.1 4.0.2 5.0.0-M1
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.glassfish.jersey.containers:jersey-container-servlet:2.34:4.0.1
========================================================================================
========================================================================================
 org.glassfish.jersey.containers:jersey-container-servlet:2.34:4.0.1 stage native-image run
========================================================================================
Starting stage 'native-image run' for 4.0.1 at 2026-05-11T07:25:11Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean test -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.glassfish.jersey.containers:jersey-container-servlet:2.34
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                 ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
        Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                              ^
  symbol:   variable LocalizationMessages

  location: class LocalizationMessagesInnerBundleSupplierTest
2 errors
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
          Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                                ^
    symbol:   variable LocalizationMessages
    location: class LocalizationMessagesInnerBundleSupplierTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
  import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                   ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

BUILD FAILED in 4s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.glassfish.jersey.containers:jersey-container-servlet:2.34 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 13s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 4.0.1 at 2026-05-11T07:25:24Z with exit code 1 after 13s
❌ ERROR: 'native-image run' for 4.0.1 failed with exit code 1.
SOFT-FAIL[native-image run][4.0.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean test -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"]
========================================================================================
 org.glassfish.jersey.containers:jersey-container-servlet:2.34:4.0.1 stage javac compile
========================================================================================
Starting stage 'javac compile' for 4.0.1 at 2026-05-11T07:25:24Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean compileTestJava -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 866ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                 ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
        Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                              ^
  symbol:   variable LocalizationMessages
  location: class LocalizationMessagesInnerBundleSupplierTest
2 errors

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
          Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                                ^
    symbol:   variable LocalizationMessages
    location: class LocalizationMessagesInnerBundleSupplierTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
  import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                   ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 4.0.1 at 2026-05-11T07:25:27Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 4.0.1 failed with exit code 1.
FAILED[javac compile][4.0.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.0.1" ./gradlew clean compileTestJava -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.0.1" ./gradlew clean compileTestJava -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25656145616/job/75305207653
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.glassfish.jersey.containers:jersey-container-servlet:2.34
Raw versions JSON: [
  "4.0.1",
  "4.0.2",
  "5.0.0-M1"
]
Parsed version count: 3
Parsed versions: 4.0.1 4.0.2 5.0.0-M1
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.glassfish.jersey.containers:jersey-container-servlet:2.34:4.0.1
========================================================================================
========================================================================================
 org.glassfish.jersey.containers:jersey-container-servlet:2.34:4.0.1 stage native-image run
========================================================================================
Starting stage 'native-image run' for 4.0.1 at 2026-05-11T07:25:06Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.0.1" ./gradlew clean test -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.glassfish.jersey.containers:jersey-container-servlet:2.34
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                 ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
        Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                              ^
  symbol:   variable LocalizationMessages
  location: class LocalizationMessagesInnerBundleSupplierTest
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
          Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                                ^

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

    symbol:   variable LocalizationMessages
    location: class LocalizationMessagesInnerBundleSupplierTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
  import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                   ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

13 actionable tasks: 5 executed, 8 up-to-dateBUILD FAILED in 4s


> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.glassfish.jersey.containers:jersey-container-servlet:2.34 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 13s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 4.0.1 at 2026-05-11T07:25:20Z with exit code 1 after 14s
❌ ERROR: 'native-image run' for 4.0.1 failed with exit code 1.
SOFT-FAIL[native-image run][4.0.1][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.0.1" ./gradlew clean test -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"]
========================================================================================
 org.glassfish.jersey.containers:jersey-container-servlet:2.34:4.0.1 stage javac compile
========================================================================================
Starting stage 'javac compile' for 4.0.1 at 2026-05-11T07:25:20Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.0.1" ./gradlew clean compileTestJava -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 873ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/build/reports/problems/problems-report.html
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                 ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
        Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                              ^
  symbol:   variable LocalizationMessages
  location: class LocalizationMessagesInnerBundleSupplierTest
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:21: error: cannot find symbol
          Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
                                ^
    symbol:   variable LocalizationMessages
    location: class LocalizationMessagesInnerBundleSupplierTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/2.34/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java:13: error: package org.glassfish.jersey.servlet.init.internal does not exist
  import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
                                                   ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 4.0.1 at 2026-05-11T07:25:23Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 4.0.1 failed with exit code 1.
FAILED[javac compile][4.0.1][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.0.1" ./gradlew clean compileTestJava -Pcoordinates="org.glassfish.jersey.containers:jersey-container-servlet:2.34"]
```

--- a/metadata/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/reachability-metadata.json
+++ b/metadata/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/reachability-metadata.json
@@ -1,0 +1,10 @@
+{
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.glassfish.jersey.servlet.internal.l10n.LocalizationMessages$BundleSupplier"
+      },
+      "bundle": "org.glassfish.jersey.servlet.internal.l10n.localization"
+    }
+  ]
+}

--- a/metadata/org.glassfish.jersey.containers/jersey-container-servlet/index.json
+++ b/metadata/org.glassfish.jersey.containers/jersey-container-servlet/index.json
@@ -61,6 +61,15 @@
     ]
   },
   {
+    "metadata-version" : "4.0.1",
+    "tested-versions" : [
+      "4.0.1"
+    ],
+    "allowed-packages" : [
+      "org.glassfish.jersey.servlet"
+    ]
+  },
+  {
     "metadata-version" : "4.0.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/glassfish/jersey/containers/jersey-container-servlet/$version$/jersey-container-servlet-$version$-sources.jar",
     "repository-url" : "https://github.com/eclipse-ee4j/jersey",

--- a/metadata/org.glassfish.jersey.containers/jersey-container-servlet/index.json
+++ b/metadata/org.glassfish.jersey.containers/jersey-container-servlet/index.json
@@ -62,6 +62,11 @@
   },
   {
     "metadata-version" : "4.0.1",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/glassfish/jersey/containers/jersey-container-servlet/$version$/jersey-container-servlet-$version$-sources.jar",
+    "repository-url" : "https://github.com/eclipse-ee4j/jersey",
+    "test-code-url" : "https://github.com/eclipse-ee4j/jersey/tree/$version$-RELEASE/tests/integration/servlet-3-init-provider/src/test",
+    "documentation-url" : "https://github.com/eclipse-ee4j/jersey/tree/$version$-RELEASE/docs/src/main/docbook",
+    "description" : "Jersey Container Servlet integrates Eclipse Jersey's Jakarta REST server runtime with Jakarta Servlet containers. It provides servlet registration, request handling, and async-context support so Jersey REST resources can run in WAR-based web applications.",
     "tested-versions" : [
       "4.0.1"
     ],

--- a/stats/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/execution-metrics.json
+++ b/stats/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/execution-metrics.json
@@ -1,0 +1,104 @@
+{
+  "fix_javac_fail:2026-05-11": {
+    "timestamp": "2026-05-11T10:08:06.785060Z",
+    "library": "org.glassfish.jersey.containers:jersey-container-servlet:4.0.1",
+    "starting_commit": "1fda8bbf7f3721472de24b000936b163f4677c67",
+    "ending_commit": "56d179014290177f333e2114bec1264ab2dbaa23",
+    "previous_library": "org.glassfish.jersey.containers:jersey-container-servlet:2.34",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "4.0.1",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 2,
+        "totalCalls": 2
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 51,
+          "missed": 3760,
+          "ratio": 0.013382,
+          "total": 3811
+        },
+        "line": {
+          "covered": 10,
+          "missed": 862,
+          "ratio": 0.011468,
+          "total": 872
+        },
+        "method": {
+          "covered": 6,
+          "missed": 230,
+          "ratio": 0.025424,
+          "total": 236
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.34",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 28,
+          "missed": 937,
+          "ratio": 0.029016,
+          "total": 965
+        },
+        "line": {
+          "covered": 4,
+          "missed": 195,
+          "ratio": 0.020101,
+          "total": 199
+        },
+        "method": {
+          "covered": 3,
+          "missed": 39,
+          "ratio": 0.071429,
+          "total": 42
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 37803,
+      "cached_input_tokens_used": 161792,
+      "output_tokens_used": 3413,
+      "iterations": 2,
+      "cost_usd": 0.1833,
+      "tested_library_loc": 10,
+      "metadata_entries": 1,
+      "previous_library_metadata_entries": 1,
+      "code_coverage_percent": 1.15,
+      "previous_library_coverage_percent": 2.01
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java",
+      "metadata_file": "metadata/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/stats.json
+++ b/stats/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "4.0.1",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 2,
+      "totalCalls" : 2
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 51,
+        "missed" : 3760,
+        "ratio" : 0.013382,
+        "total" : 3811
+      },
+      "line" : {
+        "covered" : 10,
+        "missed" : 862,
+        "ratio" : 0.011468,
+        "total" : 872
+      },
+      "method" : {
+        "covered" : 6,
+        "missed" : 230,
+        "ratio" : 0.025424,
+        "total" : 236
+      }
+    }
+  } ]
+}

--- a/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/.gitignore
+++ b/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/build.gradle
+++ b/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.glassfish.jersey.containers:jersey-container-servlet:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/gradle.properties
+++ b/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.glassfish.jersey.containers:jersey-container-servlet:4.0.1
+library.version = 4.0.1
+metadata.dir = org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/

--- a/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/settings.gradle
+++ b/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.glassfish.jersey.containers.jersey-container-servlet_tests'

--- a/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java
+++ b/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java
@@ -10,7 +10,7 @@ import java.util.Locale;
 import java.util.ResourceBundle;
 
 import org.glassfish.jersey.internal.l10n.Localizable;
-import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
+import org.glassfish.jersey.servlet.internal.l10n.LocalizationMessages;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;

--- a/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java
+++ b/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/LocalizationMessagesInnerBundleSupplierTest.java
@@ -1,0 +1,28 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_glassfish_jersey_containers.jersey_container_servlet;
+
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+import org.glassfish.jersey.internal.l10n.Localizable;
+import org.glassfish.jersey.servlet.init.internal.LocalizationMessages;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class LocalizationMessagesInnerBundleSupplierTest {
+    @Test
+    void localizableMessageExposesServletInitializationResourceBundle() {
+        Localizable message = LocalizationMessages.localizableJERSEY_APP_NO_MAPPING("SampleApplication");
+
+        ResourceBundle bundle = message.getResourceBundle(Locale.ROOT);
+
+        assertThat(bundle.getString("jersey.app.no.mapping"))
+                .isEqualTo("The Jersey servlet application, named {0}, has no servlet mapping.");
+    }
+}

--- a/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/ThreadLocalInvokerTest.java
+++ b/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/ThreadLocalInvokerTest.java
@@ -30,7 +30,7 @@ public class ThreadLocalInvokerTest {
             final ThreadLocalInvoker<Function<String, String>> invoker) {
         return (Function<String, String>) Proxy.newProxyInstance(
                 ThreadLocalInvokerTest.class.getClassLoader(),
-                new Class<?>[] { Function.class },
+                new Class<?>[] {Function.class},
                 invoker);
     }
 }

--- a/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/ThreadLocalInvokerTest.java
+++ b/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/src/test/java/org_glassfish_jersey_containers/jersey_container_servlet/ThreadLocalInvokerTest.java
@@ -1,0 +1,36 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_glassfish_jersey_containers.jersey_container_servlet;
+
+import java.lang.reflect.Proxy;
+import java.util.function.Function;
+
+import org.glassfish.jersey.servlet.internal.ThreadLocalInvoker;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ThreadLocalInvokerTest {
+    @Test
+    void proxyDelegatesMethodInvocationToThreadLocalInstance() {
+        ThreadLocalInvoker<Function<String, String>> invoker = new ThreadLocalInvoker<>();
+        invoker.set(name -> "Hello, " + name);
+
+        Function<String, String> proxy = createFunctionProxy(invoker);
+
+        assertThat(proxy.apply("Jersey")).isEqualTo("Hello, Jersey");
+    }
+
+    @SuppressWarnings("unchecked")
+    private static Function<String, String> createFunctionProxy(
+            final ThreadLocalInvoker<Function<String, String>> invoker) {
+        return (Function<String, String>) Proxy.newProxyInstance(
+                ThreadLocalInvokerTest.class.getClassLoader(),
+                new Class<?>[] { Function.class },
+                invoker);
+    }
+}

--- a/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/user-code-filter.json
+++ b/tests/src/org.glassfish.jersey.containers/jersey-container-servlet/4.0.1/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.glassfish.jersey.servlet.**"
+    },
+    {
+      "includeClasses" : "org_glassfish_jersey_containers.jersey_container_servlet.**"
+    }
+  ]
+}

